File: LeetCode/Easy/976 - Largest Perimeter Triangle/976 - Largest Perimeter Triangle.js
/**
 * @param {number[]} nums
 * @return {number}
 */
 var largestPerimeter = function(nums) {
    nums.sort((a, b) => a - b);
    
    const n = nums.length;
    for (let i = n - 1; i - 2 >= 0; --i) {
        if (nums[i] < nums[i - 1] + nums[i - 2]) {
            return nums[i] + nums[i - 1] + nums[i - 2];
        }
    }
    
    return 0;
};

File: LeetCode/Easy/976 - Largest Perimeter Triangle/976 - Largest Perimeter Triangle.cpp
class Solution {
public:
    int largestPerimeter(vector<int>& nums) {
        sort(nums.rbegin(), nums.rend());
        int n = nums.size();
        for (int i = 2; i < n; ++i) {
            if (nums[i - 2] < nums[i - 1] + nums[i]) {
                return nums[i - 2] + nums[i - 1] + nums[i];
            }
        }
        return 0;
    }
};